pkg/controller/podautoscaler/horizontal.go

```diff
@@ -370,6 +370,27 @@ func (a *HorizontalController) reconcileKey(key string) error {
 	return a.reconcileAutoscaler(hpa)
 }
 
+// return true , nil ,nil if hpav1 is  newly created for same ScaleTargetRef
+// return false, newlyHpa, nil, if hpav1 is not newly created for same ScaleTargetRef
+// newlyHpa is a hpa object which created after hpav1 and ref same ScaleTargetRef
+func (a *HorizontalController) isLasterHpa(hpav1 *autoscalingv1.HorizontalPodAutoscaler) (bool, *autoscalingv1.HorizontalPodAutoscaler, error) {
+	allHpas, err := a.hpaLister.HorizontalPodAutoscalers(hpav1.Namespace).List(labels.Everything())
+	if err != nil {
+		return false, nil, fmt.Errorf("failed to get all hpas in namespaces %v", hpav1.Namespace)
+	}
+
+	for _, h := range allHpas {
+		if h.Spec.ScaleTargetRef.Kind == hpav1.Spec.ScaleTargetRef.Kind &&
```

Comment:
I'd refactor this using an affectation `h.Spec.ScaleTargetRef` and `hpav1.Spec.ScaleTargetRef`